### DISCUSSION TITLE
REPL or print function

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#1773] [Ideas]
It seems there is no way to get the value of an variable or an expression.

```elm
let foo = 5 + 5
```
Now how do I know what is stored inside the `foo` variable?

I would love to have a `print` function so I could do:
```elm
let foo = 5 + 5
print foo
```

Or have a REPL where I could type things and it would evaluate them.
```pycon
>>> 10 + 5
15
>>> let foo = 5 + 5
>>> foo
10
```

**Comment by max-sixty** [comment]
Currently PRQL only compiles to SQL; it can't actually execute much code itself 

(it can actually execute `10 + 5`, but that's not that useful on its own...).

So I don't think this is that easy; it would need a DB integration. I'll leave it open for any ideas though!

**Reply by vanillajonathan** [reply]
Yeah, I had a feeling that it would only compile to SQL. Theoretically a database could implement native support for PRQL though.

What got me thinking was the [null handing](https://prql-lang.org/book/language-features/null.html#null-handling) chapter in the book that states that "null == null evaluates to true" and that "null != null evaluates to false" so I wanted a way to test that for myself.

I was also thinking it would be a useful feature to debug code and better understand what my variables contain.

**Reply by max-sixty** [reply]
Yes agree.

Check out the playground — that uses DuckDB, so you can evaluate small queries on some tables there.

The layout isn't that clear, we'd like to improve it.

**Reply by vanillajonathan** [reply]
I've been playing around with the playground a bit, but I've not been able to evaluate expressions and get the results of what is inside a variable in any other way than when using them as part of a query. It seems a `from` is required.

**Reply by max-sixty** [reply]
Yes this is true at the moment. You can do something like this. We'd like to make this much better. Possibly we allow a `from null` or allow no `from` (feel free to open an issue if this is something you'd like)

<img width="848" alt="image" src="https://user-images.githubusercontent.com/5635139/218283342-82e56263-c773-41cb-ab67-7c7f15ea47a2.png">
